### PR TITLE
Adding initial doc for func subscribe

### PR DESCRIPTION
**Pull request by matzew**
## Proposed Changes 

- Initial documentation for `func subscribe`

### CODE REVIEW

**Comment by matzew** [comment]
@lkingland here is my initial doc PR

File: docs/snippets/proc-subscribe-function.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ```
+
+    To subscribe the function to events for the default broker, run the following command:
+
+    ```bash
+    kn func subscribe --filter type=com.example --filter extension=my-extension-value
+    ```
+
+!!! note
+    When invoking `func deploy` the CLI will create Knative Triggers for the function.
```
Comment (dsimansk) [inline docs/snippets/proc-subscribe-function.md:36]: I'd move this note above the code blocks, line no. 7 of the `.md` file. Otherwise looks OK to me.

Comment (matzew) [reply docs/snippets/proc-subscribe-function.md]: I've separated that now, and added a more dedicated section, with a smalll focus on the deployment.

Moving the note to far up seems a bit odd to me...

**Comment by dsimansk** [comment]
/approve
/lgtm

@matzew @lkingland adding both labels & hold to merge on demand. :)
/hold

File: docs/snippets/proc-subscribe-function.md [hunk 2]
```diff
@@ -31,14 +31,3 @@ and a Knative Broker as the source of events, from where they are consumed.
     ```bash
     kn func subscribe --filter type=com.example --filter extension=my-extension-value
     ```
-
-!!! note
-    When invoking `func deploy` the CLI will create Knative Triggers for the function.
-
```
Comment (dsimansk) [inline docs/snippets/proc-subscribe-function.md]: @matzew have you intentionally removed the note box?

Comment (matzew) [reply docs/snippets/proc-subscribe-function.md]: yes, I have. please see the rendered page

**Review by lkingland — APPROVED** [review]
Looks clean and to the point  👍🏻 ❤️ Thanks for the docs!

**Pull request merged**